Scrivener experiments

nnmrts opened the pull request:
This PR will be my initial refactoring with some groundwork for Papyrus laid out.

Scrivener07 committed:
Create README.md

Scrivener07 committed:
Move script build root to support EFM namespace

Scrivener07 committed:
Added code for basic data messaging from Papyrus to AS3

Scrivener07 committed:
Added support for basic item description

See video per Discord (invite only server) https://discord.com/channels/830436661736243230/1017155572447068310/1305515181186748508

Scrivener07 committed:
Event handlers for Cassiopeia

nnmrts commented on README.md:
````diff
@@ -0,0 +1,102 @@
+PapyrusCompiler.exe "MyProject.ppj" --output="x:\some\other\directory"
+```
+
+### Developer Console
+In the event of a soft-lock or intention to debug, it is possible to open/close a menu by name using the in-game developer console.
+
+#### Open a menu by name.
+```
+HideMenu FavoritesMenu
+```
+
+#### Close a menu by name.
+```
+ShowMenu FavoritesMenu
+```
+
+#### Find the name of all currently open menus.
+```
+PAM
+```
````
Don't know if `FCF` is needed, but we could add information about it and `reloadscript EFM` as well.

Scrivener07 replied:
I added a note about `ReloadScript` from memory. I hesitated to document `FCF` and `HLP` because I'm not convinced the commands are enabled for Starfield like previous games.

Scrivener07 committed:
Update README.md

Scrivener07 committed:
Reduced batch/shell spam and added `deploy.bat` for `interface/`

Added two switches to `ROBOCOPY` which suppress a lot of verbose information.
- `/NFL` : No File List - don't log file names.
- `/NDL` : No Directory List - don't log directory names.

Scrivener07 committed:
Update README.md

Scrivener07 committed:
Refactored AS3 source into import packages

Scrivener07 committed:
Update .gitignore with pattern for any RECOVER_ folder

Update .gitignore with pattern for any RECOVER_ folder. This folder is an automatically generated backup where the folder name is includes the FLA file name, such as `RECOVER_FavoritesMenu/` or `RECOVER_Fonts_en/`.

Scrivener07 committed:
Added the vanilla fonts_en.swf source assets and code

Added the vanilla fonts_en.swf source assets and code for RSL linkage. The RSL part still needs to be configured in `FavoritesMenu.fla/`. New Report.txt is to make sure adding the font source didnt disturb anything.

Scrivener07 committed:
Updated Font symbol to use RSL linkage to `fonts_en.swf`

Updated Font symbol to use RSL linkage to `fonts_en.swf` instead of directly embedding the font into FavoritesMenu. Embedded fonts which are already provided via RSL is bad practice and should be avoided. Also some info for README.md

Scrivener07 committed:
Refactoring for Flash document library

Scrivener07 committed:
Refactoring for Flash document library

Scrivener07 committed:
Re-hide the symbol 40 (ItemItem_BoundClip)

I forgot to reset the alpha to zero after messing with it.

nnmrts committed:
add plugin file
adapt build process

nnmrts committed:
debug array persistence

nnmrts committed:
switch to flash 11 for json support

Scrivener07 commented:
I made issue #3 to be addressed at some point.

nnmrts committed:
revert a86628e

Scrivener07 committed:
Merge pull request #5 from nnmrts/equip-loop-hack

refactor build process

Scrivener07 committed:
Fixed issue #3 and finished restoration

The xmls changes are done. All library symbols are now organized and have a proper name. Same for layers. Every layer has a single shape/object. The layer name will always be the stage instance name of whatever is on the layer. If the object does not have an instance name, then a one/word descriptive name is given. If its so basic or obvious what the layer content is, I just named it by the Library symbol type, like `Shape` or `TextField`. I also fixed issue 3. Ill commit/PR/close in a soon.
